actionview/lib/action_view/template.rb

```diff
@@ -324,8 +325,11 @@ def handle_render_error(view, e) #:nodoc:
       end
 
       def locals_code #:nodoc:
+        locals = @locals.to_set - Module::DELEGATION_RESERVED_METHOD_NAMES
+        locals = locals.select { |key| /\A[a-z_][a-zA-Z_0-9]*\z/.match?(key) }
```

Comment:
Here we go: https://github.com/ruby/ruby/blob/46bbb62bed20e93b651a9f19c87634d2732f05a5/parse.y#L5184

`/\A(?![A-Z0-9])(?:[[:alnum:]_]|[^\0-\177])+\z/`